request/random.go

```
@@ -354,6 +359,87 @@ func (b *requestGetPodLogBuilder) Build(cli rest.Interface) Requester {
 	}
 }
 
+type requestPatchBuilder struct {
+	version         schema.GroupVersion
+	resource        string
+	resourceVersion string
+	namespace       string
+	name            string
+	patchType       apitypes.PatchType
+	body            interface{}
+	maxRetries      int
+}
+
+func getPatchType(patchType string) apitypes.PatchType {
+	switch patchType {
+	case "json":
+		// JSON Patch: exact field edits (RFC 6902), array of ops
+		//   [{"op": "replace", "path": "/spec/replicas", "value": 3}]
+		return apitypes.JSONPatchType
+	case "merge":
+		// Merge Patch: partial object, simple merge (RFC 7386)
+		// - Input is a partial object (e.g., {"spec": {"replicas": 2}})
+		return apitypes.MergePatchType
+	case "strategic":
+		// Strategic Merge: smart merge for native K8s types
+		// - Input is a partial object (like merge patch)
+		return apitypes.StrategicMergePatchType
+	default:
```

This may cause a "silent failure". E.g. when the string has a typo, e.g. 'jsn', the intention is using json, instead of failing, it fallback to the default. A better approach is to return an err when it's none of the expected value.